### PR TITLE
Update PDF Exporter with Images

### PR DESCRIPTION
**Pull request by anuradhas248**
When we export a framework with a inline images to a PDF the images will actually display. 
As per the discussion, we didn't write acceptance test case for this story 
Story ID: KMS-3884
Git Issue: [493](https://github.com/opensalt/opensalt/issues/493)

---
This change is [Reviewable](https://reviewable.io/reviews/opensalt/opensalt/492)

### CODE REVIEW

File: src/Controller/PdfExportController.php [hunk 1]
```diff
@@ -30,9 +30,18 @@ public function exportPdfAction(int $id): StreamedResponse
         $section = $phpWordObject->addSection();
 
         $response = $this->forward('App\Controller\Framework\CfPackageController:exportAction', ['id' => $id, '_format' => 'json']);
+        $data_array=json_decode($response->getContent(), true);
```
Comment (JosephLeon) [inline src/Controller/PdfExportController.php]: Format correctly, spacing around =

File: src/Controller/PdfExportController.php [hunk 2]
```diff
@@ -30,9 +30,18 @@ public function exportPdfAction(int $id): StreamedResponse
...
+        for($i=0; $i < count($data_array['CFItems']); ++$i)
+        {
+            $data_array['CFItems'][$i]['fullStatement']= $this->render_images($data_array['CFItems'][$i]['fullStatement']);
```
Comment (JosephLeon) [inline src/Controller/PdfExportController.php]: Formatting

File: src/Controller/PdfExportController.php [hunk 3]
```diff
@@ -30,9 +30,18 @@ public function exportPdfAction(int $id): StreamedResponse
...
+            if(isset($data_array['CFItems'][$i]['notes']))
+            {
+                $data_array['CFItems'][$i]['notes']= $this->render_images($data_array['CFItems'][$i]['notes']);
```
Comment (JosephLeon) [inline src/Controller/PdfExportController.php]: Formatting

**Comment by anuradhas248** [comment]
@roverwolf - I have updated code as per your suggestions. Please verify and confirm.

**Comment by anuradhas248** [comment]
@JosephLeon - I have taken care of your formatting comments. Please verify.

File: src/Controller/PdfExportController.php [hunk 4]
```diff
@@ -69,8 +69,8 @@ function () use ($phpWordObject) {
      */
     public function render_images($string)
     {
-        $string = str_replace('![alt text](', '<br/><br/><img src ="', $string);
-        $result = str_replace(')', '"/>', $string);
+        $pattern = '/\!\[([^\]]*)alt text\]\(*\((\b(?:(?:https?|ftp):\/\/|www\.)[-a-z0-9+&@#\/%?=~_|!:,.;]*[-a-z0-9+&@#\/%=~_|])\)/';
```
Comment (roverwolf) [inline src/Controller/PdfExportController.php]: This still requires the text for the to have `alt text` at the end of the text, it should be able to be anything.

I would suggest that only the https scheme be allowed (or none specified to allow attachments located on the same server).
  - Using http will make browsers complain about mixed content;
  - and ftp is not supported as of [Chrome 63](https://www.chromestatus.com/feature/5709390967472128) and will not be supported as of [Firefox 61](https://bugzilla.mozilla.org/show_bug.cgi?id=1404744).

So, allowing things of both these forms:
- !\[OpenSALT favicon](https://opensalt.net/favicon32.png)
- !\[GitHub Fluid-Icon](/fluidicon.png)

i.e., something along the lines of `/\!\[([^\]]*)\]\(((?:https:\/\/|\/)[^\)]+)\)/`

**Comment by roverwolf** [comment]
For [line 72](https://github.com/opensalt/opensalt/pull/492/files/df5466acfc4bc80dd09052207b615087367242a1..59d869f1c9ffa9dc09644c3d5c5e946d70f05129#diff-aa42c790ef792a6bf930fe788130454dR72) of src/Controller/PdfExportController.php (render_images method):

This still requires the text for the to have `alt text` at the end of the text, it should be able to be anything.

I would suggest that only the https scheme be allowed (or none specified to allow attachments located on the same server).
  - Using http will make browsers complain about mixed content;
  - and ftp is not supported as of [Chrome 63](https://www.chromestatus.com/feature/5709390967472128) and will not be supported as of [Firefox 61](https://bugzilla.mozilla.org/show_bug.cgi?id=1404744).

So, allowing things of both these forms:
- !\[OpenSALT favicon](https://opensalt.net/favicon32.png)
- !\[GitHub Fluid-Icon](/fluidicon.png)

i.e., something along the lines of `/\!\[([^\]]*)\]\(((?:https:\/\/|\/)[^\)]+)\)/`

**Comment by anuradhas248** [comment]
@roverwolf  - I did require changes. Could you please confirm it.

**Pull request merged**